[fix bug 1458340] Add FB videochat funnelcake banner to /firefox/.

jpetto opened the pull request:
## Description

Adds a banner with a download button for funnelcake 134 (win/win64 only) to `/en-US/firefox/` for visitors matching the following:

- referring URL includes `facebook.com`
- en-US
- Windows, IE/Edge

## Issue / Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1458340
https://bugzilla.mozilla.org/show_bug.cgi?id=1452807

## Testing

Ensure UI looks good on IE 9+ and Edge.

https://www-demo4.allizom.org/en-US/firefox/

To activate organically, you must follow a link to the above from facebook.com using either Internet Explorer or Edge. I tested by creating a private post in FB with the link above, then clicking it using IE.

You can force the banner to appear with the following URL:

https://www-demo4.allizom.org/en-US/firefox/?test-banner=true

For both testing methods above, clicking the "close" X at the top right of the banner will prevent the banner from displaying again for 7 days. This being the case, I suggest testing in a private window (or manually clearing your cookies after dismissing).

jpetto commented:
Currently blocked on GA, but code can be reviewed now.

bensternthal commented:
Closes https://github.com/mozilla/bedrock/issues/5634

P.S. References are magical and we should use them!

stephaniehobson commented:
Should I be able to test the funnelcake? I don't get facebook after downloading and installing.

stephaniehobson commented:
Also, Firefox is predicting 1d and 57minutes to download the IE10 VM so... that's not happening today.

jpetto commented on bedrock/firefox/templates/firefox/home.html:
```diff
@@ -19,10 +19,39 @@
+          <div class="header-content">
+            <p>Facebook Video Messenger calls are easy with</p>
+
+            {{ high_res_img('logos/firefox/logo-quantum-wordmark-large.png', {'alt': '', 'width': '185', 'height': '69', 'class': 'banner-logo'}) }}
```
Ah, good catch.

jpetto commented:
Added some fixes, and added an `.env` file for demo4 to allow it to serve the funnelcake. (I tested the install off demo4 and it did open `facebook.com` on first run.)

Also tested in IE 10 - had the same sticky nav issue as IE 9 - both should be fixed in this latest update.

stephaniehobson commented:
IE10/Win7 ✅

stephaniehobson commented:
IE9/Win7  ✅

jpetto commented:
@stephaniehobson Hopefully you have time to review the final GA commit before your EOD. 😬

jpetto commented:
@alexgibson I got this final commit in a bit late for Stephanie to review before her PTO - can you give it a look tomorrow?

alexgibson commented:
> @alexgibson I got this final commit in a bit late for Stephanie to review before her PTO - can you give it a look tomorrow?

The GA changes lgtm! 

Not a blocker, but one non-GA related performance tweak here would be to only load the FB images when the banner is shown. Currently, every visitor get's to download them even if they are not on IE/Edge or being referred from FB. How long is this banner going to be around for?

jpetto commented:
@alexgibson Good call on delaying image loading. Just added a commit to address.

The banner will be around until we get enough data to know if targeting this segment of visitors is worth it. If so, we'll likely keep it (or something similar) around long term.
